app/src/main/kotlin/compiler/parser/Parser.kt

```diff
@@ -0,0 +1,32 @@
+package compiler.parser
+
+import compiler.common.dfa.state_dfa.DfaState
+import compiler.common.diagnostics.Diagnostics
+import compiler.parser.grammar.AutomatonGrammar
+import compiler.parser.grammar.Grammar
+import compiler.parser.grammar.Production
+
+class Parser<S : Comparable<S>>(
+    val automatonGrammar: AutomatonGrammar<S>,
+    val nullable: Set<S>,
+    val first: Map<S, Set<S>>,
+    val follow: Map<S, Set<S>>,
+    val diagnostics: Diagnostics
+) {
+    val parseActions: Map<Pair<DfaState<S, Production<S>>, S>, Action<S>>
```

Comment:
The key type should be `Triple<Dfa<S, Production<S>>, DfaState<S, Production<S>>, S>`, unless the states of distinct DFAs are guaranteed to be different. This is currently not the case for `RegexDfaState`, which is a `data class`, and so it is possible for two DFAs to coincidentally have the same state (in the sense of the `equals` method).